### DISCUSSION TITLE
UGS on Arduino Nano clone - z-limit switch not working

### DISCUSSION THREAD

**Discussion by HubertJH** [winder/Universal-G-Code-Sender#2421] [Help] [answered]
### Version

2.1.2

### Hardware / Firmware

GRBL 1.1

### What happened

On my Arduino Nano (clone) the z-axis limit switch (D12) is not recognised by UGS (pin goes low but no alarm in UGS). I don't have an "original" Nano on hand so I cannot test whether this is related to the clone.
On the Arduino Uno all works fine.

Has anybody experienced the same issue and could shine some light on this?

I might also get an original Nano for a test - will post any findings here.

### Operating System

MacOS 14.2.1

**Comment by breiler** [comment]
Transferred this issue to a discussion as it is technically not an issue or bug in UGS.

Just to make sure that you are actually using Grbl 1.1, can you post the output from the command `$I`.

**Reply by HubertJH** [reply]
Thanks for chiming in Joacim - your work is highly appreciated!

Here is the output from $I:
>>> $I
[VER:1.1h.20190830:]
[OPT:V,15,128]
ok

I should have pointed out that I am using the exact same grbl code on both the Uno and Nano.

**Reply by HubertJH** [reply, marked as answer]
Oops! Just by chance I had a closer look at the board and it turns out that pin12 was badly soldered with no connection to the board itself. Touching it with the hot iron did the trick.

Guess that's the price you pay when using clones...

**Reply by breiler** [reply]
Cool, so problem solved now, it is triggering with a pin status?

**Reply by HubertJH** [reply]
Yep - works fine now.
